client/app/bundles/App/components/RestaurantList.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import Restaurant from './Restaurant'
+import Filter from './Filter'
+import $ from 'jquery'
+
+class RestaurantList extends React.Component{
+  constructor(props) {
+    super();
+    this.state = {
+      restaurants: [],
+      genres: ['Any'],
+      speeds: ['Any'],
+      ratings: ['Any'],
+      selected_genre: 'Any',
+      selected_speed: 'Any',
+      selected_rating: 'Any'
+    };
+  }
+  componentWillMount() {
+    let _this = this
+    $.get('restaurants/').done(function(response){
+      _this.setState({
+        restaurants: response,
+        genres: _this._uniqueProperty(response, 'genre'),
+        speeds: _this._uniqueProperty(response, 'delivery_time'),
+        ratings: _this._uniqueProperty(response, 'rating')
+      })
+    })
+  }
+  render() {
+    let selected_genre = this.state.selected_genre;
+    let selected_speed = this.state.selected_speed;
+    let selected_rating = this.state.selected_rating;
+
+    let restaurants = []
+    this.state.restaurants.forEach(function(restaurant) {
+      if (['Any',restaurant.genre].includes(selected_genre) &&
+          ['Any',restaurant.delivery_time].includes(selected_speed) &&
+          ['Any',String(restaurant.rating)].includes(String(selected_rating))) {
+        restaurants.push(<Restaurant key={restaurant.id} {...restaurant} />)
+      }
+    });
+
+    if (restaurants.length===0){
```

Comment:
Would be a good time to follow our Styleguide, i.e `array.length === 0` or `!array.length`